L1Trigger/L1THGCal/src/be_algorithms/HGCalClusteringImpl.cc

```diff
@@ -74,33 +74,28 @@ void HGCalClusteringImpl::clusterizeDR( const std::vector<edm::Ptr<l1t::HGCalTri
            ToBeFixed: if a tc is not a seed, but could be potencially be part of a cluster generated by a late seed, 
                       the tc will not be clusterized  
         */
-        int iclu=0;
-        vector<int> tcPertinentClusters; 
-        for( const auto& clu : clustersTmp){
-            if( this->isPertinent(**tc, clu, dr_) ){
-                tcPertinentClusters.push_back(iclu);
-            }
-            ++iclu;
-        }
-        if( tcPertinentClusters.empty() && isSeed[itc] ){
-            clustersTmp.emplace_back( *tc );
-        }
-        else if ( !tcPertinentClusters.empty() ){
-         
-            unsigned minDist(300);
-            unsigned targetClu(0);
-                        
-            for( int iclu : tcPertinentClusters){
-                double d = clustersTmp.at(iclu).distance(**tc);
-                if( d < minDist ){
-                    minDist = d;
-                    targetClu = iclu;
-                }
-            } 
 
-            clustersTmp.at(targetClu).addConstituent( *tc );                    
+	HGCalDetId tcDetId( (**tc).detId() );
+	int z_side = tcDetId.zside();
+
+	double minDist = dr_;
+	int targetClu = -1;
+
+	for(unsigned iclu=0; iclu<clustersTmp.size(); iclu++){
+
+	  HGCalDetId cluDetId( clustersTmp.at(iclu).detId() );
+	  if( z_side*cluDetId.zside()<0) continue;
```

Comment:
Why not using the `isPertinent()` function as before?

Reply:
Because otherwise another loop is needed to determine among the pertinent clusters which one is the closest
Here the minDist is directly stored while looping over all the clusters.

Reply:
`isPertinent()` is just doing the `z_side*cluDetId.zside()<0` check + additional distance, subdet, layer checks. So the condition could just be replaced with `!isPertinent(...)` without an additional loop.

Reply:
Agreed